code/controllers/subsystem/processing/goap.dm

```diff
@@ -0,0 +1,22 @@
+PROCESSING_SUBSYSTEM_DEF(goap)
```

Comment:
this doesnt need to be a processing subsystem since you're making your own fire() to call goap_process() instead of process()

Reply:
So should it just be a normal subsystem?

Reply:
You could just replace goap_process() with datum's defined process() instead as it's basically the same thing, if you were to replace with process() be sure not to call parent proc as that will stop the goap datum from processing

https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/processing/processing.dm#L33

Reply:
I'm going to keep this the same for debugging reasons, it's a good idea to be able to keep track of how many AIs are running GOAP and how much performance its using

Reply:
You should just make it SUBSYSTEM_DEF(goap) and move it to subsystem/goap.dm
you're not using process() [and also you can't as this is currently designed]

Reply:
wouldn't I then have to store a global list of GOAP agents instead of just doing START_PROCESSING in New for goap agents?

Reply:
But then wouldn't I have to store a global list of GOAP agents rather than doing START_PROCESSING in the new for goap agents? unless the start processing macro works for normal subsystems too

Reply:
it's debatable if you should be using START_PROCESSING because thats for process() and sets DF_PROCESSING which is not really true in this case, and may conflict with things that use process() and goap_process()